### ISSUE TITLE
Revise exits Julia

### ISSUE THREAD

**Issue by Lord-Keynes1** [timholy/Revise.jl#519]
Hello,

I've just updated Julia to 1.5 and when I do `using Revise`, Julia exits for some reason. There is no error message at all. I'm using Juno.

versioninfo():

Julia Version 1.5.0
Commit 96786e22cc (2020-08-01 23:44 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = "C:\Users\Administrator\AppData\Local\atom\app-1.49.0\atom.exe"  -a
  JULIA_NUM_THREADS = 12

**Comment by timholy** [comment]
What's your `Revise` version?

**Comment by Lord-Keynes1** [comment]
v1.0.3

**Comment by Lord-Keynes1** [comment]
I updated it to 2.7.3 and the issue disappears. I thought I had the latest version, since I just added the package. I did `] update` and it updated it to 2.7.3.

**Comment by timholy** [comment]
There's a subtle issue in that Revise 2.x is not compatible with CodeTracking 1 and LoweredCodeUtils 1 (they are intended for Revise 3, which isn't finished yet). Unfortunately Pkg's resolver likes getting two projects at newer versions even if that means it has to downgrade Revise.

You might consider pinning Revise for the next month or so.

**Comment by dlfivefifty** [comment]
Can we leave this issue open for now? I just hit the same issue where `] add Revise` defaulted to an old version, and almost opened a new issue.

**Comment by timholy** [comment]
I guess we could, but traditionally issues are open in the package that needs to fix the issue. Here there's nothing for Revise to fix, at least not until I find time to get v3 out (#497).

Here, the issue really lies with Pkg or any other packages (Juno?) that depend on Revise's dependencies.

**Comment by KristofferC** [comment]
Well, we could add compat for older revise version for the Julia version they claim support.

**Comment by timholy** [comment]
https://github.com/JuliaRegistries/General/pull/18983

**Comment by DilumAluthge** [comment]
> JuliaRegistries/General#18983

Merged.

**Comment by timholy** [comment]
I doubt those bounds were harmful, but the problem is that in the Revise 1.x days CodeTracking and LoweredCodeUtils didn't even exist. So it's happy to choose CodeTracking v1 and LoweredCodeUtils v1 even if it has to downgrade to Revise v1.

**Comment by DilumAluthge** [comment]
Hmmm. Do you want to revert those bounds?

**Comment by timholy** [comment]
I don't think it's necessary, they rationalized DataStructures/OrderedCollections at the least.

Currently recommended workaround:
```
pkg> add Revise@2
pkg> pin Revise
```
